### PR TITLE
Complete decoupling filters of eligibility trace and firing rate

### PR DESCRIPTION
**Pull request by akorgor**

### CODE REVIEW

**Comment by akorgor** [comment]
@JesusEV Overall this looks good to me. What do you think about renaming `e_bar_kappa, e_bar_kappa_reg` to `e_bar, e_bar_reg`?

**Comment by JesusEV** [comment]
@akorgor Thanks, that is a good suggestion

File: models/eprop_iaf_adapt.cpp [hunk 1]
```diff
@@ -432,6 +433,7 @@ eprop_iaf_adapt::compute_gradient( const long t_spike,
   double z_current_buffer = 1.0; // buffer containing the spike that triggered the current integration
   double psi = 0.0;              // surrogate gradient
   double L = 0.0;                // learning signal
+  double firing_rate_reg = 0.0;  // firing rate regularization term
```
Comment (akorgor) [inline models/eprop_iaf_adapt.cpp]:
```suggestion
  double firing_rate_reg = 0.0;  // firing rate regularization
```

File: models/eprop_iaf_psc_delta.cpp [hunk 2]
```diff
@@ -476,6 +477,7 @@ eprop_iaf_psc_delta::compute_gradient( const long t_spike,
   double z_current_buffer = 1.0; // buffer containing the spike that triggered the current integration
   double psi = 0.0;              // surrogate gradient
   double L = 0.0;                // learning signal
+  double firing_rate_reg = 0.0;  // firing rate regularization term
```
Comment (akorgor) [inline models/eprop_iaf_psc_delta.cpp]:
```suggestion
  double firing_rate_reg = 0.0;  // firing rate regularization
```

File: models/eprop_synapse.h [hunk 3]
```diff
@@ -319,9 +319,12 @@ class eprop_synapse : public Connection< targetidentifierT >
   //! Low-pass filtered spiking variable.
   double z_bar_ = 0.0;
 
-  //! Low-pass filtered eligibility trace.
+  //! Low-pass filtered eligibility trace with kappa as decay factor.
```
Comment (akorgor) [inline models/eprop_synapse.h]:
```suggestion
  //! Low-pass filtered eligibility trace.
```
`kappa` and `kappa_reg` are not known in the synapse. This information might be confusing here.

File: models/eprop_synapse.h [hunk 4]
```diff
@@ -319,9 +319,12 @@ class eprop_synapse : public Connection< targetidentifierT >
...
   double e_bar_ = 0.0;
 
+  //! Low-pass filtered eligibility trace with kappa_reg as decay factor.
```
Comment (akorgor) [inline models/eprop_synapse.h]:
```suggestion
  //! Low-pass filtered eligibility trace for firing rate regularization.
```

File: nestkernel/node.h [hunk 5]
```diff
@@ -824,7 +824,8 @@ class Node
    * @param t_spike_previous [in] Time of the previous spike.
    * @param z_previous_buffer [in, out] Value of presynaptic spiking variable from previous time step.
    * @param z_bar [in, out] Filtered presynaptic spiking variable.
-   * @param e_bar [in, out] Filtered eligibility trace.
+   * @param e_bar [in, out] Filtered eligibility trace with kappa as decay factor.
+   * @param e_bar_reg [in, out] Filtered eligibility trace with kappa_reg as decay factor.
```
Comment (akorgor) [inline nestkernel/node.h]:
```suggestion
   * @param e_bar [in, out] Filtered eligibility trace.
   * @param e_bar_reg [in, out] Filtered eligibility trace for firing rate regularization.
```

**Review by akorgor — COMMENTED** [review]
Only some last minor comments. Apart from these all looks good!

**Pull request merged**